.github/workflows/dev-checks.yml

```diff
@@ -31,12 +39,238 @@ jobs:
-        if: ${{ !startsWith(env.SOURCE_BRANCH,'TS-') }}
+          echo "GITHUB_TOKEN=${{ secrets.CIPIPELINE_NEXUS_PASSWORD  }}" >> $GITHUB_ENV
+      
+      - name: Check Format
+        env:
+          PR_TITLE: ${{ github.event.pull_request.title }}
+        run: |
+          if echo "${{ env.SOURCE_BRANCH }}" | grep -Eq '^revert-'; then
+              echo "Revert branch, skipping Format check"
+              exit 0
+          fi
+          if echo "${{ env.SOURCE_BRANCH }}" | grep -Eq '^(TS-|SRE-|TDEVOPS-|TOPS-)'; then
+            echo "Branch name starts with 'TS-' or 'SRE-' or 'TDEVOPS-' or 'TOPS-'"
+          else
+            echo "Branch name does not start with 'TS-' or 'SRE-' or 'TDEVOPS-' or 'TOPS-'. Checking PR title format."
+            PULL_REQUEST_TITLE="$PR_TITLE"
+            PATTERN="(TS|TDEVOPS|TOPS|SRE)-[0-9]+\s\|\s"
+            if [[ ! $PULL_REQUEST_TITLE =~ $PATTERN ]]; then
+              echo "Error: Pull request title is not in the required format. Please use 'TS-XXXX | --' or 'TDEVOPS-XXXX | --' or 'TOPS-XXXX | --' or 'SRE-XXXX | --' format."
+              exit 1
+            else 
+              echo "PR Title is in the valid format."
+            fi  
+          fi
+
+      - name: Check convoy.yaml
+        id: convoy_check
+        run: |
+          if [ -f "convoy.yaml" ]; then
+            echo "convoy_exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "convoy_exists=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install shyaml
+          echo ${{steps.convoy_check.outputs.convoy_exists}}
+
+      - name: Check double commit
+        env: 
+          PR_TITLE: ${{github.event.pull_request.title}}
+          BRANCH: ${{ github.event.pull_request.head.ref}}
+        run: |
+          if ([[ '${{ github.event.pull_request.user.login }}' != 'cipipelinetessell' ]]) && ([[ "$BRANCH" == *"double_commit"* ]] || [[ "$PR_TITLE" == *"Double Commit"* ]]); then
+            if ([[ "$BRANCH" == *"revert"* ]] && [[ "$PR_TITLE" == *"Revert"* ]]); then
+              echo "Revert Double commit Branch. Allowed"
+            else   
+              echo "Exclude Double commit naming in Branch, PR title and try again."
+              exit 1
+            fi
+          else
+            echo "No double commit conflicts found in Branch or PR title."
+          fi
+
+      - name: import packages for json lint
+        run: |
+          sudo sudo npm install jsonlint -g
+          
+      - name: Check for Jsonlint
+        run: |
+          set -e
+          git diff --name-only --diff-filter=AM ${{ github.event.pull_request.base.sha }}...${{ github.event.pull_request.head.sha }} | grep '\.json$' | xargs -I{} sh -c 'jsonlint -q "{}" || { echo "::error::Invalid JSON file: {}"; exit 1; }'
+
+      - name: install octokit/rest
```

Comment:
can we not have all `npm install *` in one step

Reply:
Json-lint needs to installed globally so not sure about that, rest I have added in single statement